code/game/machinery/doors/door.dm

```diff
@@ -256,20 +256,16 @@
 	update_icon()
 	if(visible && !glass)
 		set_opacity(TRUE)
-		if(filler)
-			filler.set_opacity(opacity)
+		if(length(filler_turfs))
+			change_filler_opacity(opacity)
 	operating = FALSE
 
 /obj/structure/machinery/door/proc/requiresID()
 	return TRUE
 
-
-/obj/structure/machinery/door/proc/update_flags_heat_protection(turf/source)
-
-
 /obj/structure/machinery/door/proc/autoclose()
-	var/obj/structure/machinery/door/airlock/A = src
-	if(!A.density && !A.operating && !A.locked && !A.welded && A.autoclose)
+	var/obj/structure/machinery/door/airlock/airlock = src
```

Comment:
this typecast should be pointless and you can just check density operating locked etc below. if you can't this proc shouldn't be a `/obj/structure/machinery/door` proc it should ee a `/obj/structure/machinery/door/airlock` proc